lib/rules/no-duplicate-imports.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @fileoverview Restrict usage of duplicate imports.
+ * @author Simen Bekkhus
+ * @copyright 2016 Simen Bekkhus. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var importsInFile = [];
+
+    return {
+        "ImportDeclaration": function(node) {
+            if (node && node.source && node.source.value) {
+                var value = node.source.value.trim();
+
+                if (importsInFile.indexOf(value) !== -1) {
+                    context.report(node, "'{{importName}}' import is duplicated.", {
```

Comment:
This overload of `report` method was deprecated.
Please use the following pattern:

```js
context.report({
    node: node,
    message: "'{{importName}}' import is duplicated.",
    data: {importName: value}
});
```